pkg/resources/database/resource_db_mysql.go

```diff
@@ -0,0 +1,286 @@
+		return
+	}
+
+	for _, db := range svc.Databases {
+		if string(db) == data.DatabaseName.ValueString() {
+			return
+		}
+	}
+	diagnostics.AddError("Client Error", "Unable to find database for the service")
+}
+
+// CreateResource creates the resource according to the model, and then
+// update computed fields if applicable
+func (data MysqlDatabaseResourceModel) CreateResource(ctx context.Context, client *v3.Client, diagnostics *diag.Diagnostics) {
+
+	createRequest := v3.CreateDBAASMysqlDatabaseRequest{
+		DatabaseName: v3.DBAASDatabaseName(data.DatabaseName.ValueString()),
+	}
+
+	op, err := client.CreateDBAASMysqlDatabase(ctx, data.Service.ValueString(), createRequest)
+	if err != nil {
+		diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to create service database, got error %s", err.Error()),
+		)
+		return
+	}
+
+	if _, err := client.Wait(ctx, op, v3.OperationStateSuccess); err != nil {
+		diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to create service database, got error %s", err.Error()),
+		)
+		return
+	}
+
+	svc, err := client.GetDBAASServiceMysql(ctx, data.Service.ValueString())
+	if err != nil {
+		diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read database service pg, got error: %s", err))
+		return
+	}
+
+	for _, db := range svc.Databases {
+		if string(db) == data.DatabaseName.ValueString() {
+			return
+		}
+	}
+	diagnostics.AddError("Client Error", "Unable to find newly created database for the service")
+}
+
+// DeleteResource deletes the resource
+func (data MysqlDatabaseResourceModel) DeleteResource(ctx context.Context, client *v3.Client, diagnostics *diag.Diagnostics) {
+
+	op, err := client.DeleteDBAASMysqlDatabase(ctx, data.Service.ValueString(), data.DatabaseName.ValueString())
+	if err != nil {
+		diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to delete service database, got error %s", err.Error()),
+		)
+		return
+	}
+	_, err = client.Wait(ctx, op, v3.OperationStateSuccess)
+	if err != nil {
+		diagnostics.AddError(
+			"Client Error",
+			fmt.Sprintf("Unable to delete service database, got error %s", err.Error()),
+		)
+		return
+	}
+
+}
+
+// UpdateResource updates the remote resource w/ the new model
+func (MysqlDatabaseResourceModel) UpdateResource(ctx context.Context, client *v3.Client, diagnostics *diag.Diagnostics) {
+	// Nothing to do as these resources are systematically recreated
+}
+
+func (data MysqlDatabaseResourceModel) WaitForService(ctx context.Context, client *v3.Client, diagnostics *diag.Diagnostics) {
+	_, err := waitForDBAASServiceReadyForFn(ctx, client.GetDBAASServiceMysql, data.Service.ValueString(), func(t *v3.DBAASServiceMysql) bool { return t.State == v3.EnumServiceStateRunning })
+	// DbaaS API is unstable when a service goes from rebuilding from running,
+	// this wait time helps avoid that
+	time.Sleep(time.Second * 10)
```

Comment:
Put the sleep value as a var in utils.go so we can bump it at one place  if needed.